src/Sarif.Multitool/FileWorkItemsCommand.cs

```diff
@@ -140,6 +135,8 @@ private bool ValidateOptions(FileWorkItemsOptions options)
 
             valid = ValidateOutputFileOptions(options) && valid;
 
+            valid = DriverUtilities.ReportWhetherOutputFileCanBeCreated(options.OutputFilePath, options.Force, fileSystem) && valid;
+
```

Comment:
I don't quite understand the option combinations. If OutputFilePath == null and options.Inline == true and options.Force == default/false, which seems valid, doesn't ReportWhetherOutputFileCanBeCreated fail? That is, it seems like the output/input file swap on line 35 should happen before the ValidateOptions() call.

Reply:
Good catch. Yes, in the other commands, I was careful to do all of the "pre-validation" option manipulation before calling `ValidateOptions`. I missed this one.

---
In reply to: [312279620](https://github.com/microsoft/sarif-sdk/pull/1625#discussion_r312279620) [](ancestors = 312279620)